### PR TITLE
ARROW-8153: [Packaging] Update the conda feedstock files and upload artifacts to Anaconda

### PR DESCRIPTION
**Pull request by kszucs**
The windows builds were failing, so the feedstock files must be updated.

Under the same hat add support for uploading the produced artifacts to Anaconda labeled as nightly.

The uploaded conda packages are available under https://anaconda.org/ursa-labs/repo

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-osx-clang-py36

**Comment by kszucs** [comment]
@github-actions crossbow submit -g conda

**Comment by kszucs** [comment]
@github-actions crossbow submit -g conda

**Comment by kszucs** [comment]
@github-actions crossbow submit -g conda

**Comment by kszucs** [comment]
@github-actions crossbow submit -g conda

**Comment by kszucs** [comment]
@github-actions crossbow submit -g conda

**Comment by kszucs** [comment]
@github-actions crossbow submit -g conda

File: dev/tasks/conda-recipes/azure.linux.yml [hunk 1]
```diff
@@ -52,3 +54,8 @@ jobs:
     env:
       CROSSBOW_GITHUB_TOKEN: $(CROSSBOW_GITHUB_TOKEN)
     displayName: Upload packages as a GitHub release
+
+  # Upload to custom anaconda channel
+  - script: |
+      anaconda -t $(CROSSBOW_ANACONDA_TOKEN) upload -l nightly --force build_artifacts/linux-64/*.tar.bz2
```
Comment (kou) [inline dev/tasks/conda-recipes/azure.linux.yml]: https://dev.azure.com/ursa-labs/crossbow/_build/results?buildId=8372&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=3367cfe7-6aa5-5a88-c888-e6814d74f6eb&l=16

I don't know why but `tar.bz2` can't be found...

File: dev/tasks/conda-recipes/azure.linux.yml [hunk 2]
```diff
@@ -59,4 +59,3 @@ jobs:
   - script: |
       anaconda -t $(CROSSBOW_ANACONDA_TOKEN) upload -l nightly --force build_artifacts/linux-64/*.tar.bz2
     displayName: Upload packages to Anaconda
-    workingDirectory: arrow/dev/tasks/conda-recipes
```
Comment (kou) [inline dev/tasks/conda-recipes/azure.linux.yml]: Ah.

Comment (kszucs) [reply dev/tasks/conda-recipes/azure.linux.yml]: Don't even mention :)

**Comment by kszucs** [comment]
@kou the builds are greeen now

File: dev/tasks/conda-recipes/arrow-cpp/bld.bat [hunk 3]
```diff
@@ -18,6 +18,7 @@ cmake -G "Ninja" ^
       -DARROW_BUILD_UTILITIES:BOOL=OFF ^
       -DARROW_BUILD_STATIC:BOOL=OFF ^
       -DCMAKE_BUILD_TYPE=release ^
+      -DARROW_SSE42:BOOL=OFF ^
```
Comment (wesm) [inline dev/tasks/conda-recipes/arrow-cpp/bld.bat]: Can you rebase and change this to be `ARROW_SIMD_LEVEL=NONE`?

Comment (kou) [reply dev/tasks/conda-recipes/arrow-cpp/bld.bat]: Done. (And sorted.)

**Comment by kou** [comment]
@github-actions crossbow submit -g conda

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**